seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/sink/CassandraSinkWriter.java

```diff
@@ -0,0 +1,144 @@
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.AsyncResultSet;
+import com.datastax.oss.driver.api.core.cql.BatchStatement;
+import com.datastax.oss.driver.api.core.cql.BoundStatement;
+import com.datastax.oss.driver.api.core.cql.ColumnDefinitions;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.type.DataType;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class CassandraSinkWriter extends AbstractSinkWriter<SeaTunnelRow, Void> {
+
+    private final CassandraConfig cassandraConfig;
+    private final SeaTunnelRowType seaTunnelRowType;
+    private final ColumnDefinitions tableSchema;
+    private final CqlSession session;
+    private BatchStatement batchStatement;
+    private List<BoundStatement> boundStatementList;
+    private List<CompletionStage<AsyncResultSet>> completionStages;
+    private final PreparedStatement preparedStatement;
+    private final AtomicInteger counter = new AtomicInteger(0);
+
+    public CassandraSinkWriter(CassandraConfig cassandraConfig, SeaTunnelRowType seaTunnelRowType, ColumnDefinitions tableSchema) {
+        this.cassandraConfig = cassandraConfig;
+        this.seaTunnelRowType = seaTunnelRowType;
+        this.tableSchema = tableSchema;
+        this.session = CassandraClient.getCqlSessionBuilder(
+            cassandraConfig.getHost(),
+            cassandraConfig.getKeyspace(),
+            cassandraConfig.getUsername(),
+            cassandraConfig.getPassword(),
+            cassandraConfig.getDatacenter()).build();
+        this.batchStatement = BatchStatement.builder(cassandraConfig.getBatchType()).build();
+        this.boundStatementList = new ArrayList<>();
+        this.completionStages = new ArrayList<>();
+        this.preparedStatement = session.prepare(initPrepareSQL());
+    }
+
+    @Override
+    public void write(SeaTunnelRow row) throws IOException {
+        BoundStatement boundStatement = this.preparedStatement.bind();
+        addIntoBatch(row, boundStatement);
+        if (counter.getAndIncrement() >= cassandraConfig.getBatchSize()) {
+            flush();
+            counter.set(0);
+        }
+    }
+
+    private void flush() {
+        if (cassandraConfig.getAsyncWrite()) {
+            completionStages.forEach(resultStage -> resultStage.whenComplete(
+                (resultSet, error) -> {
+                    if (error != null) {
+                        error.printStackTrace();
+                    }
+                }
+            ));
+            completionStages.clear();
+        } else {
+            try {
+                this.session.execute(this.batchStatement.addAll(boundStatementList));
+            } catch (Exception e) {
+                for (BoundStatement statement : boundStatementList) {
```

Comment:
Suggest output the Exception through log.

Reply:
Thanks for your advice. I'll fix it.